flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/abilities/source/WatermarkPushDownSpec.java

```diff
@@ -127,10 +129,18 @@ public DefaultWatermarkGeneratorSupplier(
                     new ArrayList<>(Arrays.asList(generatedWatermarkGenerator.getReferences()));
             references.add(context);
 
+            String code = generatedWatermarkGenerator.getCode();
+            String splitCode =
+                    JavaCodeSplitter.split(
```

Comment:
Can we take this into `GeneratedClass`? We have 20+ `JavaCodeSplitter.split`... We can just pass a TableConfig to `GeneratedClass`.